### PR TITLE
Various changes and improvements

### PR DESCRIPTION
**Pull request by Panquesito7**
- Add [Factorio Luacheck](https://github.com/marketplace/actions/factorio-mod-luacheck).
- Various `.luacheckrc` fixes.
- Add build status badge.
   - It will be available until merged.

After merging, you must update `check-release.yml` to point to `minetest-mods`, otherwise, build checks won't work correctly.

```
luacheckrc_url: https://raw.githubusercontent.com/minetest-mods/lavastuff/master/.luacheckrc
```
Currently, it's pointing to my fork, so I can verify if build checks are correct.

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -1,8 +1,9 @@
 unused_args = false
 allow_defined_top = true
+max_line_length = 999
```
Comment (LoneWolfHT) [inline .luacheckrc]: Why 999 and not 120 (which my VSCode plugin may have been set to check for)?

File: .github/workflows/check-release.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+name: Check & Release
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: lint
+        uses: Roang-zero1/factorio-mod-luacheck@master
+        with:
+          luacheckrc_url: https://raw.githubusercontent.com/Panquesito7/lavastuff/master/.luacheckrc # Will be changed soon to "minetest-mods"
```
Comment (LoneWolfHT) [inline .github/workflows/check-release.yml]: Github is complaining about the lack of a newline at the end of the file

**Comment by LoneWolfHT** [comment]
Thanks!
Let me know when this is ready for merging, and please list any extra steps I'll need to take, never done anything like this before

File: .luacheckrc (hunk 1 above)
Comment (Panquesito7) [reply .luacheckrc]: `999` is enough if you're going to have long lines.
It's similar to having unlimited lines.

**Comment by Panquesito7** [comment]
> Let me know when this is ready for merging

Ready for merge.

> and please list any extra steps I'll need to take

No further steps needed, just follow the steps above. 🙂

File: .luacheckrc (hunk 1 above)
Comment (LoneWolfHT) [reply .luacheckrc]: Ok, I guess that's fine then

**Comment by LoneWolfHT** [comment]
Looks like it worked. Thanks again

**Pull request merged**